gourmet/gtk_extras/TextBufferMarkup.py

```diff
@@ -78,26 +80,35 @@ def __init__ (self):
         GObject.GObject.__init__(self)
 
     def set_text (self, txt):
+        if isinstance(txt, bytes):
+            # data loaded from the database are bytes, not str
+            txt = txt.decode()
         Gtk.TextBuffer.set_text(self,"")
         try:
-            self.parsed,self.txt,self.separator = Pango.parse_markup(txt,'\x00')
-        except:
-            print('Problem encountered escaping text: "%s"'%txt)
+            self.parsed, self.attrList, self.txt, self.separator = Pango.parse_markup(txt, -1, '\x00')
+        except Exception as e:  # unescaped text (g-markup-error-quark), eg. contains &amp
+            print(f"Problem encountered escaping text: {txt}: {e}")
             import traceback; traceback.print_exc()
             txt=xml.sax.saxutils.escape(txt)
-            self.parsed,self.txt,self.separator = Pango.parse_markup(txt,'\x00')
-        self.attrIter = self.parsed.get_iterator()
+            self.parsed, self.attrList, self.txt, self.separator = Pango.parse_markup(txt, -1, '\x00')
+
+        self.attrIter = self.attrList.get_iterator()
```

Comment:
Documentation says, we should destroy this iterator once we're done with it (http://lazka.github.io/pgi-docs/#Pango-1.0/classes/AttrList.html#Pango.AttrList.get_iterator)